libraries/botbuilder/src/cloudAdapter.ts

```diff
@@ -216,7 +229,7 @@ export class CloudAdapter extends CloudAdapterBase implements BotFrameworkHttpAd
 
         // Create server
         const server = new WebSocketServer(
-            await new NodeWebSocketFactory().createWebSocket(req, socket, head),
+            await this.webSocketFactory.createWebSocket(req, socket, head),
```

Comment:
In order to minimize the changes to public interface, would the following implementation work?
- Create a protected method (say, `CreateWebSocketFactory()`) in CloudAdapter that returns a `NodeWebSocketFactoryBase`.
- The default implementation will return a new instance of the `NodeWebSocketFactory`.
- This method can then be overridden if a different factory is desired.

This would eliminate the need to change the constructor signature, and keep the changes minimal. Please let me know your thoughts on this proposal.